### PR TITLE
Operators as provided methods on Observable instead of extension traits

### PR DESCRIPTION
**Pull request by M-Adoo**
This pr removed all extension traits of operators  and provided methods on Observable. 

@helgoboss I think you mentioned in #68  make sense:

> I noticed you prefer observable::create over Observable::create. There's nothing wrong with that and it looks clean to me but it is different from probably most other Rx implementations where create is a static method on the main Observable class (even in RxCpp although in C++ free factory functions are usually preferred over static class factory methods). I think the idea there is to have a kind of main class/trait (Observable) and make most things accessible from there - which improves API convenience. Just an observation, I don't have any hard opinion on that.

Do you have time to do it  after this pr merged ?

### CODE REVIEW

File: guide/developing.md [hunk 1]
```diff
@@ -2,17 +2,20 @@
 
 ## To implement a operator tip?
```
Comment (helgoboss) [inline guide/developing.md]:
```suggestion
## How to implement an operator type
```

File: guide/developing.md [hunk 2]
```diff
@@ -2,17 +2,20 @@
...
 
-`rxRust` has two base trait named `Observable` and `SharedObservable`, these two trait almost the same. `Observable` is used for local thread, and `SharedObservable` is a thread safe version. So your should remember implement both two for your operator.
+- Add a method in `Observable` to construct your operator type.
+- Implement `Observable` for your operator type.
+- implement `LocalObservable` for your operator type to subscribe upstream and not care thread safe.
```
Comment (helgoboss) [inline guide/developing.md]:
```suggestion
- Implement `LocalObservable` for your operator type in order to handle a subscribe (upstream) for single-threaded use
```

File: guide/developing.md [hunk 3]
```diff
@@ -2,17 +2,20 @@
...
+- implement `SharedObservable` for your operator type to subscribe upstream and keep thread safe.
```
Comment (helgoboss) [inline guide/developing.md]:
```suggestion
- Implement `SharedObservable` for your operator type in order to handle subscription (upstream) for multi-threaded use
```

File: guide/developing.md [hunk 4]
```diff
@@ -2,17 +2,20 @@
...
 
 ## How to implement a creation operator?
 
-Creation operators are functions that can be used to create an Observable with some common predefined behavior. `rxRust` already have a generic struct named ObservableBase<Emitter> for implement creation operators.
+Creation operators are functions that can be used to create an Observable with some common predefined behavior. `rxRust` already have a generic struct named ObservableBase<LocalEmitter> for implement creation operators.
```
Comment (helgoboss) [inline guide/developing.md]:
```suggestion
Creation operators are functions that can be used to create an Observable with some common predefined behavior. `rxRust` already has a generic struct named `ObservableBase<LocalEmitter>` for implementing creation operators.
```

File: guide/developing.md [hunk 5]
```diff
@@ -2,17 +2,20 @@
...
 
-ObservableBase<Emitter> assume those Observables are accept a `Subscriber`  to emit values. So a easy way to implement a creation operator is :
+ObservableBase<LocalEmitter> assume those Observables are accept a `Subscriber`  to emit values. So a easy way to implement a creation operator is :
```
Comment (helgoboss) [inline guide/developing.md]:
```suggestion
`ObservableBase<LocalEmitter>` assumes those Observables accept a `Subscriber`  to emit values. So an easy way to implement a creation operator is :
```

File: guide/developing.md [hunk 6]
```diff
@@ -2,17 +2,20 @@
...
 
-- first, implement a `XxxEmitter` type which implement a `Emitter` trait.
+- first, implement a `XxxEmitter` type which implement a `LocalEmitter` trait.
```
Comment (helgoboss) [inline guide/developing.md]:
```suggestion
- First, implement an `XxxEmitter` type which implements a `LocalEmitter` trait.
```

File: guide/developing.md [hunk 7]
```diff
@@ -2,17 +2,20 @@
...
 - second, implement the function you want, this function just return a concrete type `ObservableBase<XxxEmitter>`.
```
Comment (helgoboss) [inline guide/developing.md]:
```suggestion
- Second, implement the function you want. It should return a concrete type `ObservableBase<XxxEmitter>`.
```

File: guide/developing.md [hunk 8]
```diff
@@ -2,17 +2,20 @@
...
-- third, like `Observable`, `Emitter` has thread safe version named `SharedEmitter`, remember implement it for your `XxxEmitter`.
+- third, like `LocalObservable`, `LocalEmitter` has thread safe version named `SharedEmitter`, remember implement it for your `XxxEmitter`.
```
Comment (helgoboss) [inline guide/developing.md]:
```suggestion
- Third, like `LocalObservable`, `LocalEmitter` has a thread-safe version named `SharedEmitter`, remember to implement it for your `XxxEmitter`.
```

**Comment by helgoboss** [comment]
> This pr removed all extension traits of operators and provided methods on Observable.

Amazing. So much progress! Added some suggestions for improved documentation.

> Do you have time to do it after this pr merged ?

Yes, I'm glad to give it a go as soon as this is merged.

**Comment by M-Adoo** [comment]
Thank you for your review,  it's very helpful.  Please see again.

**Comment by helgoboss** [comment]
> > I noticed you prefer observable::create over Observable::create. There's nothing wrong with that and it looks clean to me but it is different from probably most other Rx implementations where create is a static method on the main Observable class (even in RxCpp although in C++ free factory functions are usually preferred over static class factory methods). I think the idea there is to have a kind of main class/trait (Observable) and make most things accessible from there - which improves API convenience. Just an observation, I don't have any hard opinion on that.
> 
> Do you have time to do it after this pr merged ?

@M-Adoo I experimented a little and think it's not possible. Because `Observable` has associated types, simply writing `Observable::create(...)` is not possible ... no matter if you make `fn create(...)` a default method of `trait Observable` or a method of `impl dyn Observable`. Do you agree? If yes, I would suggest to leave things as they are.

**Comment by M-Adoo** [comment]
@helgoboss I agree. Let it go. 😂

**Pull request merged**